ext/extralite/database.c

```diff
@@ -386,6 +386,59 @@ VALUE Database_prepare(VALUE self, VALUE sql) {
   return rb_funcall(cPreparedStatement, ID_NEW, 2, self, sql);
 }
 
+/* call-seq:
+ *   Database.status(op, need_watermark, reset)
+ *
+ * Returns sqlite3 global status in the form of op_value or [ op_value, op_highwatermark ]
+ * or throws an exception. All arguments are optional.
+ */
+VALUE Database_singleton_status(int argc, VALUE* argv, VALUE self) {
+  VALUE op, wmrk, reset;
+  sqlite3_int64 cur, hwm;
+
+  rb_scan_args(argc, argv, "03", &op, &wmrk, &reset);
+
+  if (!RTEST(op)) op = INT2FIX(SQLITE_STATUS_MEMORY_USED);
+  int rc = sqlite3_status64(NUM2INT(op), &cur, &hwm, RTEST(reset) ? 1 : 0);
+  if (rc != SQLITE_OK) rb_raise(cError, "%s", sqlite3_errstr(rc));
+
+  // we don't check if it is RB_FIXABLE
+  if (!RTEST(wmrk)) return LONG2FIX(cur);
+
+  VALUE res = rb_ary_new2(2);
+  rb_ary_push(res, LONG2FIX(cur));
+  rb_ary_push(res, LONG2FIX(hwm));
+
+  return res;
+}
+
+/* call-seq:
+ *   db.status(op, need_watermark, reset)
+ *
+ * Returns database state status in the form of op_value or [ op_value, op_highwatermark ]
+ * or throws an exception. Last 2 arguments are optional.
+ */
+VALUE Database_status(int argc, VALUE* argv, VALUE self) {
```

Comment:
Same here, the method signature should be `Database#status(op, reset_flag) #=> [current, high_water]`.

Reply:
implemented